action.yaml

```diff
@@ -0,0 +1,366 @@
+    default: ""
+  skip-desc-branch:
+    description: Skip the DESCRIPTION file modification for the feature branch.
+    required: false
+    default: false
+  skip-desc-dev:
+    description: Skip the DESCRIPTION file modification for the development branch.
+    required: false
+    default: false
+  skip-install:
+    description: Skip the installation of R package dependencies.
+    required: false
+    default: false
+
+branding:
+  icon: 'download'
+  color: 'orange'
+
+runs:
+  using: composite
+  steps:
+
+  - name: Get branch names
+    id: branch-names
+    uses: tj-actions/branch-names@v8
+
+  - name: Install required packages
+    uses: r-lib/actions/setup-r-dependencies@v2
+    with:
+      packages: |
+        any::sessioninfo
+      extra-packages: |
+        any::pkgdepends
+        any::cli
+        any::desc
+      install-pandoc: false
+
+  - name: Prepare DESCRIPTION file (feature branch)
+    if: >
+      inputs.skip-desc-branch == 'false' ||
+      steps.branch-names.outputs.is_default == 'true'
+    id: desc-branch
+    shell: Rscript {0}
+    run: |
+      cat("::group::Define utils\n")
+      base_packages <- c("R", "base", "utils", "stats", "graphics", "grDevices", "datasets", "methods", "tools", "parallel", "compiler")
+      desc_field_create <- function(d, key) {
+        d$set_list(key, character(0))
+      }
+      desc_field_append <- function(d, key, value) {
+        old_values <- d$get_field(key)
+        new_values <- if (identical(trimws(old_values), "")) {
+          value
+        } else {
+          paste(old_values, value, sep = ", ")
+        }
+        d$set_list(key, new_values)
+      }
+      desc_field_clear_if_empty <- function(d, key) {
+        if (d$get_field(key) == "") {
+          d$del(key)
+        }
+      }
+      find_ref <- function(package, refs) {
+        res <- pkgdepends::parse_pkg_refs(refs) |>
+          Filter(
+            x = _,
+            f = function(ref) {
+              ref$package == package
+            }
+          )
+        if (length(res) == 0) {
+          return(NULL)
+        }
+        if (length(res) > 1) {
+          warning(sprintf("Multiple refs found for package %s. Only the first one will be used.", package))
+        }
+        res <- res[[1]]
+        if (isFALSE(is(res, "remote_ref_github"))) {
+          warning(sprintf("Ref for package %s is not a GitHub type of ref.", package))
+          return(NULL)
+        }
+        return(res)
+      }
+      remote_ref_add_branch <- function(remote_ref, branch_name) {
+        pkgdepends::parse_pkg_ref(
+          sprintf(
+            "%s@%s",
+            gsub(
+              "//@.*",
+              "",
+              remote_ref$ref
+            ),
+            branch_name
+          )
+        )
+      }
+      remote_ref_ping <- function(remote_ref) {
+        x <- pkgdepends::new_pkg_deps(remote_ref$ref, config = list(dependencies = FALSE))
+        x$solve()
+        x$get_solution()$status == "OK"
```

Comment:
This seems like the best path forward to check for github and non-github repositories.

However, if all we want to do is support github, then `{gh}` might be faster

```r
resp <- gh::gh_gql(
  '
  query($owner: String!, $repo: String!, $ref: String!) {
    repository(owner: $owner, name: $repo) {
      ref(qualifiedName: $ref) {
        id
      }
    }
  }
  ',
  variables = list(
    owner = remote_ref$username,
    repo = remote_ref$repo,
    ref = sprintf("refs/heads/%s", remote_ref$commitish)
  )
)

!is.null(resp$data$repository$ref$id)
```

Reply:
Actually I started with what you showed here :)
The reason why I switched into `pkgdepends` is to limit the dependencies of this functionality. A developer / maintainer just needs to get familiar with `pkgdepends` only - without GH graphql and its API in R.